src/transformation/visitors/call.ts

```diff
@@ -284,18 +287,39 @@ export const transformCallExpression: FunctionVisitor<ts.CallExpression> = (node
     }
 
     const callExpression = lua.createCallExpression(callPath, parameters, node);
-    if (optionalContinuation && isContextualCall) {
+    if (optionalContinuation && !isContextualCall) {
         optionalContinuation.contextualCall = callExpression;
     }
     return wrapResultInTable ? wrapInTable(callExpression) : callExpression;
 };
 
-function isContextualCallExpression(context: TransformationContext, signature: ts.Signature | undefined): boolean {
-    const declaration = signature?.getDeclaration();
-    if (!declaration) {
-        return !context.options.noImplicitSelf;
+function isContextualCallExpression(
```

Comment:
You seem to have inverted the meaning of this function -> is contextual call == has non-void context, currently you changed it to return the opposite of that

Reply:
Yeah, I tried to make the function a bit more readable, as nearly all returns were reversed in the previous function.
I guess that does make the name of it incorrect though, so I'll go ahead and change it back.